src/client/grpc_client.erl

```diff
@@ -761,13 +781,18 @@ call(Process, Request, Timeout) ->
             erlang:demonitor(Mref, [flush]),
             Reply;
         {'DOWN', Mref, _, _, Reason} ->
-            exit(Reason)
+            call_result(ExitOnCrash, Reason)
     after Timeout ->
               erlang:demonitor(Mref, [flush]),
               {error, {grpc_utils:codename(?GRPC_STATUS_DEADLINE_EXCEEDED),
                        <<"Waiting for response timeout">>}}
     end.
 
+call_result(_ExitOnCrash = true, Reason) ->
+    exit(Reason);
+call_result(_ExitOnCrash = false, Reason) ->
+    error({exit, Reason}).
```

Comment:
Q: shouldn't this be:

```suggestion
    {error, {exit, Reason}}.
```

so that an error tuple is returned?

Reply:
Other functions already raise errors on unexpected behaviour.
We need error instead of exit because in the client code we don't want to block own shutdown signals,  but want to catch ones from the called process.